pubsublite/ps/settings.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package ps
+
+import (
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"cloud.google.com/go/pubsublite/internal/wire"
+
+	pb "google.golang.org/genproto/googleapis/cloud/pubsublite/v1"
+)
+
+const (
+	// MaxPublishRequestCount is the maximum number of messages that can be
+	// batched in a single publish request.
+	MaxPublishRequestCount = wire.MaxPublishRequestCount
+
+	// MaxPublishMessageBytes is the maximum allowed serialized size of a single
+	// Pub/Sub message in bytes.
+	MaxPublishMessageBytes = wire.MaxPublishMessageBytes
+
+	// MaxPublishRequestBytes is the maximum allowed serialized size of a single
+	// publish request (containing a batch of messages) in bytes.
+	MaxPublishRequestBytes = wire.MaxPublishRequestBytes
+)
+
+// KeyExtractorFunc is a function that extracts an ordering key from a Message.
+type KeyExtractorFunc func(*pubsub.Message) []byte
+
+// PublishMessageTransformerFunc transforms a pubsub.Message to a Pub/Sub Lite
+// PubSubMessage. If this returns an error, the pubsub.PublishResult will be
+// errored and the PublisherClient will consider this a fatal error and
+// terminate.
+type PublishMessageTransformerFunc func(*pubsub.Message) (*pb.PubSubMessage, error)
+
+// PublishSettings control the batching of published messages. These settings
+// apply per partition.
+//
+// Use DefaultPublishSettings for defaults, as an empty PublishSettings will
+// fail validation.
+type PublishSettings struct {
+	// Publish a non-empty batch after this delay has passed. Must be > 0.
+	DelayThreshold time.Duration
```

Comment:
I see that these names match what's in the "wire" package, but I wonder if they should include "batching" in their names.

Reply:
I used identical names to CPS [pubsub.PublishSettings](https://godoc.org/cloud.google.com/go/pubsub#PublishSettings) (the `wire` settings were previously in the `pubsublite` package, and thus also inherited these names). Agree that having "batching" in the name makes it clearer and might rename the fields for the `wire` package in a separate PR - although I might err on the side of consistency, have yet to decide :)